dm/worker/server.go

```diff
@@ -674,21 +676,50 @@ func (s *Server) startWorker(cfg *config.MysqlConfig) error {
 		return err
 	}
 	for _, kv := range resp.Kvs {
-		infos := common.UpstreamSubTaskKeyAdapter.Decode(string(kv.Key))
-		taskName := infos[1]
 		task := string(kv.Value)
-		cfg := config.NewSubTaskConfig()
-		if err = cfg.Decode(task); err != nil {
+		subTaskcfg := config.NewSubTaskConfig()
+		if err = subTaskcfg.Decode(task); err != nil {
+			return err
+		}
+		subTaskcfg.LogLevel = s.cfg.LogLevel
+		subTaskcfg.LogFile = s.cfg.LogFile
+
+		subTaskCfgs = append(subTaskCfgs, subTaskcfg)
+	}
+
+	if len(cfg.RelayBinLogName) == 0 && len(cfg.RelayBinlogGTID) == 0 {
+		minPos, err := getMinPosInAllSubTasks(subTaskCfgs)
+		if err != nil {
 			return err
 		}
-		cfg.LogLevel = s.cfg.LogLevel
-		cfg.LogFile = s.cfg.LogFile
+		if minPos != nil {
```

Comment:
We do not support GTID now, add some TODO here?

Reply:
add in 736aa48